Blog post for Nim Devroom at Fosdem 2022 (#323)
nim-lang/website: jekyll/_posts/2022-02-04-nim-devroom-fosdem-2022.md

---

> title: "Nim Devroom at Fosdem 2022"

Reviewer: ```suggestion
title: "Nim Devroom at FOSDEM 2022"
```

author: pietroppeter, pmunch, dom96
excerpt: "Tomorrow (Sat, Feb 5th) there is a Nim Devroom at FOSDEM 2022"
---

> ## Nim Devroom at FOSDEM 2022

Reviewer 2: Maybe the title should include date of the devroom?
```suggestion
## Nim Devroom at FOSDEM 2022 this Saturday 5th Feb
```

> We are happy to announce that all preparations for Nim Devroom at Fosdem 2022 are now complete and

Reviewer: ```suggestion
We are happy to announce that all preparations for the Nim Devroom at FOSDEM 2022 are now complete and
```

> we are looking forward to 8 excellent talks with more than 5 hours of Nim related content. The conference is online this year

Reviewer: ```suggestion
we are looking forward to 8 excellent talks with more than 5 hours of Nim-related content. The conference is online this year
```

> and the talks  will stream live tomorrow on the Fosdem website. This is the final schedule for the Nim Devroom which will be

Reviewer: ```suggestion
and the talks will stream live tomorrow on the FOSDEM website. This is the final schedule for the Nim Devroom, which will be
```

held tomorrow (Saturday, February 5th) from 12:30 to 17:40: https://fosdem.org/2022/schedule/track/nim_programming_language/

> **Note that all times are Brussels' times (CET).**

Reviewer 2: Nitpick (you'd write "London time" not "London's time" so the `'` isn't necessary)

```suggestion
**Note that all times are Brussels time (CET).**
```

Reviewer: Alternatively, maybe something like:

> **Note that all times are Central European Time (UTC+1)**.

which helps a reader who isn't familiar with CET, or is unsure about how CET interacts with daylight savings time.

Author: I will go with the suggestion by @ee7

> FOSDEM is a two-day event organised by volunteers to promote the widespread use of free and open source software.

Reviewer: ```suggestion
FOSDEM is a two-day event organised by volunteers to promote the widespread use of free and open-source software.
```

Here, open-source is a compound adjective that describes software.

See e.g. https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/o/open-source

It normally takes place in Brussels but this year (like last year) it will be held remotely. Participation is open to anyone
and there is no need to register. This year is the first year that Nim has a dedicated devroom and we want to send special
thanks to our speakers for their contributions!

You will be able to interact with the speakers through a Matrix chat and at the end of every talk there will be a live Q&A with
the speaker. You may use a Matrix Client ([Element](https://element.io) is good choice that you can use both from web and as an
app) and use the link provided in the page of any talk, or you can access the chat at this link:
https://chat.fosdem.org/#/room/#nim-devroom:fosdem.org

Are you ready to:

* Hear the [past, present and future of concurrency](https://fosdem.org/2022/schedule/event/nim_concurrency/)

>   in Nim directly from the mouth of our BDFL;

Reviewer: ```suggestion
  in Nim directly from the mouth of our BDFL
```

>   in Nim directly from the mouth of our BDFL

Reviewer 2: "From the mouth" reads a little odd

```suggestion
  in Nim directly from our BDFL
```

Reviewer 2: Actually, thinking about it some more. Maybe instead of "BDFL" just write "Nim's creator". BDFL is a tounge in cheek thing and probably shouldn't be referred to in official posts like this :)

Author: I will go with creator

* Learn why Nim is well positioned for [Next generation micro-controller programming](https://fosdem.org/2022/schedule/event/nim_ngmicrocontrollers/)

>   thanks to the efforts of our own PMunch;

Reviewer: ```suggestion
  thanks to the efforts of our own PMunch
```

* Discover that a Nim based open source library ([LibVF.io](https://github.com/Arc-Compute/libvf.io))

>   is being used to [bring vGPU functionality to normal consumer GPUs](https://fosdem.org/2022/schedule/event/nim_libvfio/)

Reviewer: The semicolon at the end of the line was already omitted here.

Author: ok, then we are going withoutsemicolons. fine for me

* See how Nim can be used to create [Polymorph,  a fast and frugal entity-component-system library](https://fosdem.org/2022/schedule/event/nim_polymorph/);
* Help a 6 year old build [Potato Zombies, a 3D game using Enu and Nim](https://fosdem.org/2022/schedule/event/nim_potatozombies/);

> * Have your mind blown by Zach showing off [Pararules, a Nim rules engine for reactive programs and games](https://fosdem.org/2022/schedule/event/nim_pararules/);

Reviewer: ```suggestion
* Have your mind blown by Zach showing off [Pararules, a Nim rules engine for reactive programs and games](https://fosdem.org/2022/schedule/event/nim_pararules/)
```

> * Join a data scientist working in genomics and follow him while he [boosts performance of his code switching from Python to Nim](https://fosdem.org/2022/schedule/event/nim_hpcfrompythontonim/);

Reviewer: ```suggestion
* Join a data scientist working in genomics and follow him while he [boosts performance of his code switching from Python to Nim](https://fosdem.org/2022/schedule/event/nim_hpcfrompythontonim/)
```

> * And finally step through the power levels of [Nim metaprogramming in the real word](https://fosdem.org/2022/schedule/event/nim_metaprogramming/), thanks to master Treeform!

Reviewer 2: ```suggestion
* And finally step through the power levels of [Nim metaprogramming in the real word](https://fosdem.org/2022/schedule/event/nim_metaprogramming/), thanks to Treeform!
```

I'd omit the "master", not disputing treeform's amazing skills but IMO it sounds off.

Reviewer: ~Perhaps "main Treeform"?~

Yeah, I'd omit "master" too. I guess it relies on a reference to "power level" via internet culture and Dragonball Z, which not every reader will have in their background.

> * And finally step through the power levels of [Nim metaprogramming in the real word](https://fosdem.org/2022/schedule/event/nim_metaprogramming/), thanks to master Treeform!

Reviewer 3: I'm leaving this uppercase for now. Let's see what others say....

> * And finally step through the power levels of [Nim metaprogramming in the real word](https://fosdem.org/2022/schedule/event/nim_metaprogramming/), thanks to master Treeform!

Reviewer: If we're happy to link to GitHub profiles, one workaround is to write [@treeform](https://github.com/treeform) (and [@PMunch](https://github.com/PMunch) above it), but then it's probably a bit weird (or even unfair) if we don't do that for everyone.

Maybe [@treeform]() without a hyperlink? :)

We cannot wait to see you there!

### Final remarks

- Make sure to check out the [FOSDEM Code of Conduct](https://fosdem.org/2022/practical/conduct/).

> - We invite to read the rest of FOSDEM program for other interesting topics in [the full list of events](https://fosdem.org/2022/schedule/events/)

Reviewer: ```suggestion
- We invite you to read the rest of FOSDEM program for other interesting topics in [the full list of events](https://fosdem.org/2022/schedule/events/).
```
